.github/workflows/build.yml

```diff
@@ -32,9 +35,14 @@ jobs:
           distribution: 'zulu'
           java-version: 19
 
-      - name: Test
+      - name: Full build
+        if: matrix.full-build
         run: ./gradlew build
 
+      - name: File system check
+        if: "!matrix.full-build"
+        run: ./gradlew :kotlinpoet:allTests
```

Comment:
Wasn't quite sure which task to run in the case we just need to check file system interactions.


```shell
./gradlew :kotlinpoet:check --dry-run
```

gives me:

```
:kotlinpoet:checkKotlinGradlePluginConfigurationErrors SKIPPED
:kotlinpoet:compileKotlinJvm SKIPPED
:kotlinpoet:compileJava SKIPPED
:kotlinpoet:jvmProcessResources SKIPPED
:kotlinpoet:jvmMainClasses SKIPPED
:kotlinpoet:compileTestKotlinJvm SKIPPED
:kotlinpoet:compileTestJava SKIPPED
:kotlinpoet:jvmTestProcessResources SKIPPED
:kotlinpoet:jvmTestClasses SKIPPED
:kotlinpoet:jvmTest SKIPPED
:kotlinpoet:allTests SKIPPED
:kotlinpoet:jvmApiBuild SKIPPED
:kotlinpoet:jvmApiCheck SKIPPED
:kotlinpoet:apiCheck SKIPPED
:spotlessInternalRegisterDependencies SKIPPED
:kotlinpoet:spotlessKotlin SKIPPED
:kotlinpoet:spotlessKotlinCheck SKIPPED
:kotlinpoet:spotlessCheck SKIPPED
:kotlinpoet:testJdk11 SKIPPED
:kotlinpoet:testJdk17 SKIPPED
:kotlinpoet:testJdk19 SKIPPED
:kotlinpoet:testJdk8 SKIPPED
:kotlinpoet:check SKIPPED
BUILD SUCCESSFUL in 980ms
```

Whereas:

```shell
./gradlew :kotlinpoet:allTests
```

gives me:

```
:kotlinpoet:checkKotlinGradlePluginConfigurationErrors SKIPPED
:kotlinpoet:compileKotlinJvm SKIPPED
:kotlinpoet:compileJava SKIPPED
:kotlinpoet:jvmProcessResources SKIPPED
:kotlinpoet:jvmMainClasses SKIPPED
:kotlinpoet:compileTestKotlinJvm SKIPPED
:kotlinpoet:compileTestJava SKIPPED
:kotlinpoet:jvmTestProcessResources SKIPPED
:kotlinpoet:jvmTestClasses SKIPPED
:kotlinpoet:jvmTest SKIPPED
:kotlinpoet:allTests SKIPPED
```

leaving out testJdk8, testJdk11 etc.

I assumed the intention is to reduce GitHub Action minutes so `kotlinpoet:allTests` is okay?

Reply:
My vote is to run `check`, so that we don't inadvertently exclude any tasks hooked into it, that are important to execute. Curious to hear from Jake though, as it's not completely clear to me which file system interactions were meant.